src/Models/Classes/ViewerConfigUtility.ts

```diff
@@ -999,129 +999,143 @@ abstract class ViewerConfigUtility {
         return value >= Number(values[0]) && value < Number(values[1]);
     }
 
-    static getMatchingRangeFromValue(
-        ranges: IValueRange[],
-        value: number
-    ): IValueRange | null {
-        let targetRange: IValueRange = null;
-        if (ranges) {
-            for (const range of ranges) {
-                if (
-                    value >= Number(range.values[0]) &&
-                    value < Number(range.values[1])
-                ) {
-                    targetRange = range;
-                }
-            }
-        }
-
-        return targetRange;
-    }
+    /**
+     * Function that determines the lengths of the gauge arcs, the color for the arcs
+     * and the scaled location of where the gauge's needle points
+     * @param valueRanges
+     * @param currentValue
+     * @returns an object that has percent (scaled location of needle point),
+     * array of colors, and array of arcs length
+     */
     static getGaugeWidgetConfiguration(
```

Comment:
One more add to this PR that could be useful. Could you add tests for these new methods?

Reply:
the stories are directly testing these methods

Reply:
Stories allow visual confirmation that these methods are working but they are manually checked. Unit tests can avoid any error/regression in an automated way. Both are useful.